### PR TITLE
Generate new Adobe Recommendation CSV feed every 4 hours.

### PR DESCRIPTION
**Pull request by Omicron7**
This generates an adobe target recommendations CSV feed every 4 hours.
Based on scored cru.org URI and data from associated AEM pages.

### CODE REVIEW

File: scheduled/cru-org-feed.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    const s3 = new AWS.S3({apiVersion: '2006-03-01'})
+
+    /**
+     * Fetches cru.org json information through querybuilder
+     * @returns {Promise<void>}
+     */
+    const getCruOrgJson = async () => {
+      return request({
+        method: 'GET',
+        url: 'http://prodpub1.aws.cru.org:4503/bin/querybuilder.json',
```
Comment (wrandall22) [inline scheduled/cru-org-feed.js]: This won't work if pub1 is down, but pub2 is up. Not sure if there is a loadbalanced URL for the publisher directly. If not maybe we could do a retry (if pub1 fails, try pub2).

File: scheduled/cru-org-feed.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      }
+    })
+
+    const csvStringify = CSV.stringify({
+      delimiter: ',',
+      columns: zipObject(feedColumns, map(feedColumns, () => ''))
+    })
+
+    const scoreStream = await getScoreStream()
+    // scoreStream.pipe(scoreTransform).pipe(csvStringify).pipe(addStreamHeader(csvHeader)).pipe(process.stdout)
```
Comment (wrandall22) [inline scheduled/cru-org-feed.js]: I think this can be removed.

File: scheduled/cru-org-feed.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    await s3.upload({
+      Bucket: 'scale-of-belief-lambda-production', // process.env[''],
```
Comment (wrandall22) [inline scheduled/cru-org-feed.js]: Should replace this with the environment var

File: scheduled/cru-org-feed.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+              record['entity.message'] = content['subtitle'].replace(new RegExp('[\\r\\n]', 'gi'), ' ').trim()
+            }
+            if (typeof content['image'] !== 'undefined' && typeof content['image']['fileReference'] !== 'undefined') {
+              record['entity.thumbnailUrl'] = record['entity.pageUrl'].replace(/\.html$/i, '/_jcr_content/image.transform/CruHalf432x243/img.png')
+            }
+          }
+          if (typeof primarycat !== 'undefined') {
+            record['entity.primary_category'] = primarycat
+          }
+          // drop last subcat as it's the article name
```
Comment (wrandall22) [inline scheduled/cru-org-feed.js]: Is this guaranteed by something, or is it just the case now?

File: serverless.yml [hunk 5]
```diff
@@ -65,6 +65,11 @@ functions:
     timeout: 30
     events:
       - sns: ${env:SNS_TOPIC_ARN}
+  cru-org-feed:
+    handler: scheduled/cru-org-feed.handler
```
Comment (wrandall22) [inline serverless.yml]: If you want, you can put `redshift.js` into scheduled as well. I can rebase my testing branch.

File: scheduled/cru-org-feed.js (hunk 4 above)
Comment (Omicron7) [reply scheduled/cru-org-feed.js]: subcats is just url split by `/`, so last part is always the page/article name.

File: scheduled/redshift.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ const {Client} = require('pg')
 const copyToSteam = require('pg-copy-streams').to
 const rollbar = require('../config/rollbar')
 const AWS = require('aws-sdk')
-const dateUtil = require('./util/date-util')
+const dateUtil = require('../db/util/date-util')
```
Comment (wrandall22) [inline scheduled/redshift.js]: This can move along with the `redshift.js` since it's the only file that uses it.

**Pull request merged**